explorer/ast/value.h

```diff
@@ -1060,6 +1060,10 @@ class ConstraintType : public Value {
              equality_constraints_, rewrite_constraints_, lookup_contexts_);
   }
 
+  auto souce_loc() const -> SourceLocation {
+    return self_binding_->source_loc();
```

Comment:
Sorry, I meant for this to be a member of `ConstraintTypeBuilder`, not of `ConstraintType`.